lib/puppet/util/windows/registry.rb

```diff
@@ -213,7 +213,7 @@ def read(key, name_ptr, *rtype)
         when Win32::Registry::REG_MULTI_SZ
           result = [ type, data_ptr.read_wide_string(string_length).split(/\0/) ]
         when Win32::Registry::REG_BINARY
-          result = [ type, data.read_bytes(0, byte_length) ]
+          result = [ type, data_ptr.read_bytes(0, byte_length) ]
```

Comment:
This doesn't actually appear to be enough to fix. 

I started to add tests, but this is private and we mock out any calls that would get to this level. I think we'll need to add some specs related to this.